### DISCUSSION TITLE
Grafana Agent Flow in jsonnet?

### DISCUSSION THREAD

**Discussion by sentoz** [grafana-cold-storage/agent#5966] [Q&A]
Hello!

Tell me, is jsonnet expected to be prepared for grafana-agent-flow deployment?

**Comment by rfratto** [comment]
Hi, it's unlikely that we'll provide full Jsonnet configs for deploying Flow mode. We've been focusing our efforts instead on the Helm chart, since Helm has much broader adoption than Jsonnet does. 

If you're looking to deploy Flow mode on Jsonnet, you can use the following:

* Our [river-jsonnet](https://github.com/grafana/agent/tree/main/operations/river-jsonnet) library to write River configs in Jsonnet. 
* Tanka's [Helm chart support](https://tanka.dev/helm) to deploy our [Helm chart](https://github.com/grafana/agent/tree/main/operations/helm/charts/grafana-agent).

The combination of these two is how we deploy Flow in our environments.

**Reply by sentoz** [reply]
Thanks for the answer.

It's a pity that you took helm as a basis, because...  it is not at all native to yaml/json.
I think a tool in which you need to specify indents for generating yaml is the height of insanity).  I hope the community comes to its senses someday)

Thanks for the advice on how to do it yourself.
